core/src/iosMain/kotlin/com/revenuecat/purchases/kmp/Purchases.ios.kt

```diff
@@ -446,4 +605,12 @@ public actual class Purchases private constructor(private val iosPurchases: IosP
 
     public actual fun setCreative(creative: String?): Unit =
         iosPurchases.setCreative(creative)
+
+    private fun isIOSVersion18OrAbove(): Boolean {
+        return currentOSVersion() > "18.0.0"
```

Comment:
It's necessary for us to check the OS version here. Despite the native functions being marked as available on iOS 18.0+, macOS 15.0+, tvOS 18.0+, watchOS 11.0+, and visionOS 2.0+, KMP allows us to call these functions regardless of the OS versions 🤠

Other hybrids skirt around this issue in two ways:
- They perform an availability check in their Obj-C/Swift wrappers
- PHC also performs an availability check

However, since we're calling the native functions directly from Kotlin, we can't take advantage of those mechanisms. This _technically_ works on iOS, but it's not great for a few reasons:
- It depends on the `UIDevice` class, which isn't available in all OS's
- It's only checking for the iOS version, ignoring the OS requirements for other OS's.

Do you happen to know of a better way to handle this @JayShortway?

Reply:
As of now, our KMP SDK only supports iOS. (See our configured compilation targets [here](https://github.com/RevenueCat/purchases-kmp/blob/main/build-logic/convention/src/main/kotlin/com/revenuecat/purchases/kmp/buildlogic/convention/configureKotlin.kt#L21).) So I think that avoids those downsides? 

(I would like the KMP SDK to support more Apple targets in the future. When we do, we can use `expect`/`actual` to make sure this works on all Apple targets. There's https://github.com/RevenueCat/purchases-kmp/issues/126 to track it.)

The comparison might be a bit brittle, as it's comparing strings, but I think it should work (Unlikely, but if `currentOSVersion` ever prefixes a 0, it won't work anymore.) 

Although shouldn't it be this? 
 ```suggestion
         return currentOSVersion() >= "18.0.0"
```